src/sharedData/components/SharedDataUpload/ShareDataLinks.js

```diff
@@ -54,7 +54,14 @@ import { groupDataEntryUploadsByStatus } from './utils';
 
 export const VALIDATION_SCHEMA = yup
   .object({
-    url: yup.string().trim().ensure().transform(transformURL).url().required(),
+    url: yup
+      .string()
+      .trim()
+      .ensure()
+      .transform(transformURL)
+      .validTld()
```

Comment:
issue: please add `validTld()` to the other places we call transformURL (GroupForm, KeyInfoStep, RichTextEditor/index.js)